resources/check-layout-th.js

```diff
@@ -208,8 +208,10 @@ var testNumber = 0;
 var highlightError = false; // displays outline around failed test element.
 var printDomOnError = true; // prints dom when test fails.
 
-window.checkLayout = function(selectorList, callDone = true)
+window.checkLayout = async function(selectorList, callDone = true)
 {
+    await new Promise(r => setTimeout(r, 50)); // Give the anchor-position polyfill some time to apply changes
```

Comment:
@jerivas This helps in most cases (except a few of the `-dynamic` tests, where I think a longer delay might be required), but I don't think this change is likely to be acceptable in upstream WPT based on what Mason was saying. I think our best bet for getting something merged upstream is going to require no changes to regular (non-polyfill) WPT tests... Maybe we can get away with something that only applies when `--inject-script` is used?

I'm happy to discuss this with you, and Mason also volunteered to be a sounding board for this.

Reply:
Yeah, I thought maintaining our own branch was a viable option, but if we want to merge with upstream then we'll need to figure out another way. Would it be okay for `checkLayout` to look for a global variable specifying the delay? That way our polyfill can set the variable and leave things alone for the "normal" case and even other polyfills that use `--inject-script`.

Reply:
That makes sense -- I think maintaining our own branch is a last resort, though.

I like the global variable (or body class-name) idea, as long as there isn't a race condition and the polyfill can add it before the WPT looks for it. But this seems like a good approach to try, and also probably a good idea to run it by Mason before we get too far?

Reply:
By global variable I mean something attached to `window`. If that doesn't work, I can try to figure out a way to apply the delay only when `--inject-script` is used

Reply:
Works for me! I'd say go ahead and run this approach by Mason to see what he thinks.

I guess if we wanted to give polyfills even more control, we could use a boolean flag and wait for the polyfill to remove it or set it to `false`.

Reply:
Interesting idea! I think it *might* be ok to land this in upstream WPT, because it doesn't affect the global testrunner, it just affects all tests that use check-layout-th.js. (Though there are quite a few of those.)

@bfgeek what do you think?

One comment on the code: don't you need to add **`await`** `checkLayout()` everywhere? Several of the tests call `checkLayout` twice, doing things between the calls.

Reply:
Also @xiaochengh for comment.

Reply:
> Hey - unfortunately I don't think we should land this upstream as it's implicitly testing that implementations can have this delay.

Can you elaborate? The idea would be that this delay is only legal when the test harness is running with injected polyfills. Otherwise (when run regularly), the delay would be asserted to be zero.

Reply:
Is the delay for the polyfill to be loaded or for the polyfill to run?

Reply:
@bfgeek for the polyfill to run. Specifically, when the tests mutate the DOM and want to assert something using the `testLayout` function, we want to give the polyfill a few ms to apply the changes.

Also, just to be clear, the fragment at the top of this thread is outdated. The actual change only applies the delay if a variable on `window` is set